src/Dialog/Dialog.js

```diff
@@ -46,6 +45,92 @@ export const styleSheet = createStyleSheet('MuiDialog', (theme) => {
   };
 });
 
+type Props = {
```

Comment:
I'm going to merge this as I've already had a rebase nightmare with it. Will hold off on more, and we can discuss any other things that need to be done on gutter and I'll get to it on Monday. Thanks for all the feedback, all good things and this code is better for it!